cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -666,38 +668,51 @@ io.cdap.cdap.runtime.spi.provisioner.ClusterStatus getClusterStatus(String name)
       .map(cluster -> convertStatus(cluster.getStatus()))
       .orElse(io.cdap.cdap.runtime.spi.provisioner.ClusterStatus.NOT_EXISTS);
 
-    // if it failed, try to get the create operation and log the error message
+    return status;
+  }
+
+  String getClusterFailureMsg(String name) throws RetryableProvisionException {
+    // TODO: this can return multiple operations if the cluster name is the same (ex: a previous cluster had the
```

Comment:
whenever we add a todo, we should also open a jira and mention it in the comment.

Reply:
I had a tracking bug. Put it back in the comment.